### PR TITLE
`Tutorial groups`: Add utilization and attendance to tutorial groups sidebar

### PR DESCRIPTION
**Pull request by rabeatwork**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

The utilization rate is an important indicator for students whose tutorial groups have been canceled due to a public holiday, for example. It allows them to easily see which alternative tutorial groups they can attend instead.

### Description

Add the Utilization (and attendance if not capacity is available) to the sidebar. 

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- 1 Tutorial Group with capacity and attendance
- 1 Tutorial Group with no capacity

1. Log in to Artemis
2. Navigate to Tutorial Groups 
3. See the capacity/average attendance for the tutorial group 

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Screenshots

Screenshot 2024-05-13 at 15 29 02

### CODE REVIEW

File: src/main/webapp/app/overview/course-overview.service.ts [hunk 1]
```diff
@@ -101,24 +105,36 @@ export class CourseOverviewService {
...
     }
     mapTutorialGroupToSidebarCardElement(tutorialGroup: TutorialGroup): SidebarCardElement {
         const tutorialGroupCardItem: SidebarCardElement = {
             title: tutorialGroup.title ?? '',
             id: tutorialGroup.id ?? '',
-            subtitleLeft: tutorialGroup.language,
-            subtitleRight: tutorialGroup.nextSession?.start?.format('MMM DD, YYYY') ? 'Next: ' + tutorialGroup.nextSession?.start?.format('MMM DD, YYYY') : 'No upcoming session',
+            subtitleLeft: tutorialGroup.nextSession?.start?.format('MMM DD, YYYY')
+                ? tutorialGroup.nextSession?.start?.format('MMM DD, YYYY')
+                : this.translate.instant('artemisApp.courseOverview.sidebar.noUpcomingSession'),
```
Comment (JohannesStoehr) [inline src/main/webapp/app/overview/course-overview.service.ts:118]:
```suggestion
            subtitleLeft: tutorialGroup.nextSession?.start?.format('MMM DD, YYYY') ?? this.translate.instant('artemisApp.courseOverview.sidebar.noUpcomingSession'),
```

File: src/main/webapp/i18n/de/student-dashboard.json [hunk 2]
```diff
@@ -42,7 +42,8 @@
                 "noDate": "Ohne Datum",
                 "all": "Übungsgruppen",
                 "registered": "Meine Übungsgruppen",
-                "further": "Weitere Übungsgruppen"
+                "further": "Weitere Übungsgruppen",
+                "noUpcomingSession": " Kein Termin"
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/de/student-dashboard.json:46]: Is the space intended? I personally think that we should add this to the html-file and not the translation, what do you think?

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
Should we also color the tutorials according to their utilization like the exercises are colored for difficulty? There is also a utilization meter in Artemis that uses color as well: 
Bildschirmfoto 2024-05-12 um 23 07 33

**Review by pzdr7 — APPROVED** [review]
Tested on TS1.

- G1 Created a tutor group; left capacity blank; registered 2 students
- G2 Created a tutor group; capacity 7; registered 2 students
- G3 Created a blank tutor group (no capacity, no students)

image

-> Works as expected! ✅

**Review by egekurt123 — APPROVED** [review]
Tested manually on TS1, works as expected

**Review by JohannesStoehr — APPROVED** [review]
Code and screenshots look good to me

**Review by pzdr7 — APPROVED** [review]
Reapprove

**Review by az108 — APPROVED** [review]
reapprove

**Review by egekurt123 — APPROVED** [review]
reapprove after small changes

**Pull request merged**